pkg/querier/queryrange/value.go

```diff
@@ -0,0 +1,73 @@
+package queryrange
+
+import (
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql"
+)
+
+// FromResult transforms a promql query result into a samplestream
+func FromResult(res *promql.Result) ([]SampleStream, error) {
+	if res.Err != nil {
```

Comment:
should you add a nil check ? not sure about the context of this when called though.

Reply:
This comes from the prometheus promql engine and should be fine:
```
type Query interface {
	// Exec processes the query. Can only be called once.
	Exec(ctx context.Context) *Result
	// Close recovers memory used by the query result.
	Close()
	// Statement returns the parsed statement of the query.
	Statement() Statement
	// Stats returns statistics about the lifetime of the query.
	Stats() *stats.QueryTimers
	// Cancel signals that a running query execution should be aborted.
	Cancel()
}
```